.travis.yml

```diff
@@ -1,8 +1,25 @@
 language: php
 php:
- - '5.4'
- - '5.5'
- - '5.6'
- - '7.0'
- - hhvm
- - nightly
+  - 5.6
+  - 7.0
+  - 7.1
+  - hhvm
+  - nightly
+
+matrix:
+  allow_failures:
+    - php: 5.6
+    - php: 7.0
+    - php: hhvm
+    - php: nightly
+  fast_finish: true
+
+before_script:
+  - composer self-update
+  - composer require php-coveralls/php-coveralls
```

Comment:
The `coveralls` is the code coverage service, but it seems that it uses the `codecov` (another codeverage service) to generate the code coverage report automatically in `after_success` step during Travis CI build.

I think this `php-coveralls` package is not necessary, and it should be removed.

Reply:
What would you offer for publishing coverage report?

Reply:
@techno-express, thank you for your reply.

I'm just confused about why this have two code coverage services.

I think we can choose one of these services..

Reply:
Two? Where? codecov getting that report created by the php-coverall script it's not a service.

Reply:
As far as I know, the `php-coveralls` is not necessary.

The codecov is a bash script and it will read the `coverage.xml` generated by `phpunit` command.

You can remove the `composer require php-coveralls/php-coveralls` to prove this.

Reply:
The Travis CI build is failed because it need to run the `composer install` before running the `vendor/bin/phpunit --coverage-clover=coverage.xml`.

The detailed build log is available [here](https://travis-ci.org/purplepixie/phpdns/jobs/565249266).

Reply:
According to the [latest build log](https://travis-ci.org/purplepixie/phpdns/builds/565272592), it's worked successfully now.